### ISSUE TITLE
Parser error, false positive, concat && in SELECT

### ISSUE THREAD

**Issue by larshp** [abaplint/abaplint#1534]
```abap
  SELECT 'SICF'              AS  main_object,
        src~icf_name && src~icfparguid          AS main_obj_name,
        src~icf_langu         AS language,
        src~icf_docu          AS description,
        'icfdocu' AS txttab
       FROM icfdocu AS src
   APPENDING CORRESPONDING FIELDS OF TABLE @ta_result
       WHERE icf_docu IN @search_range
         AND icf_langu     LIKE @language.
```
